core/src/Streamly/Internal/FileSystem/Path/Common.hs

```diff
@@ -269,63 +280,148 @@ hasDrive a
     | not (isAlpha (unsafeIndexChar 0 a)) = False
     | otherwise = True
 
--- | On windows, the path starts with a separator.
-isCurDriveRelativeLocation :: (Unbox a, Integral a) => Array a -> Bool
-isCurDriveRelativeLocation a =
+-- | A path that starts with a alphabet followed by a colon e.g. @C:...@.
+hasDrive :: (Unbox a, Integral a) => Array a -> Bool
+hasDrive a = Array.byteLength a >= 2 && unsafeHasDrive a
+
+-- | A path that contains only an alphabet followed by a colon e.g. @C:@.
+isDrive :: (Unbox a, Integral a) => Array a -> Bool
+isDrive a = Array.byteLength a == 2 && unsafeHasDrive a
+
+------------------------------------------------------------------------------
+-- Relative or Absolute
+------------------------------------------------------------------------------
+
+-- | A path relative to cur dir it is either @.@ or starts with @./@.
+isRelativeCurDir :: (Unbox a, Integral a) => Array a -> Bool
+isRelativeCurDir a
     -- Assuming the path is not empty.
-    isSeparator Windows (wordToChar (Array.getIndexUnsafe 0 a))
+    | wordToChar (Array.getIndexUnsafe 0 a) /= '.' = False
```

Comment:
This is unsafe. We don't check the length before indexing.

Reply:
Can be written as `wordToChar (Array.getIndexUnsafe 0 a) == '.'`
It's more easier to understand.

Reply:
> This is unsafe. We don't check the length before indexing.

It assumes path is not empty. This assumption is there at some places already. If it is legal to have non-empty path can you raise an issue to fix these?